File: tests/src/main/java/com/github/klee0kai/test/di/base_phone/qualifiers/PhoneOsVersion.java
package com.github.klee0kai.test.di.base_phone.qualifiers;

import java.util.Objects;

public class PhoneOsVersion {
    public String version;

    public PhoneOsVersion(String version) {
        this.version = version;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PhoneOsVersion that = (PhoneOsVersion) o;
        return Objects.equals(version, that.version);
    }

    @Override
    public int hashCode() {
        return Objects.hash(version);
    }
}

File: tests/src/main/java/com/github/klee0kai/test/di/base_phone/identifiers/PhoneOsVersion.kt
package com.github.klee0kai.test.di.base_phone.identifiers

import java.util.*

class PhoneOsVersion(@JvmField var version: String?) {
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as PhoneOsVersion
        return version == that.version
    }

    override fun hashCode(): Int {
        return Objects.hash(version)
    }
}
